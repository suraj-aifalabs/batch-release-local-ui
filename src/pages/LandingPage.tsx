import CustomButton from "@/custom_components/CustomButton"
import CustomContainer from "@/custom_components/CustomContainer"
import { Link } from "react-router-dom"
import { Input } from "@/components/ui/input"
import { useState } from "react"
const LandingPage = () => {
    const [username, setUsername] = useState("")
    const [email, setEmail] = useState("")
    return (
        <CustomContainer >
            <div className="flex flex-col justify-center h-[90dvh] gap-4 items-center w-full">
                <Input type="username" placeholder="Username" className="w-[300px]" value={username} onChange={(e) => setUsername(e.target.value)} />
                <Input type="email" placeholder="Email" className="w-[300px]" value={email} onChange={(e) => setEmail(e.target.value)} />
                <Link state={{ username, email }} to="/setting/release"><CustomButton type="primary" text="Login" /></Link>
            </div >
        </CustomContainer>
    )
}

export default LandingPage
