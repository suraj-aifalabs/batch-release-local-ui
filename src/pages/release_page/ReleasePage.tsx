import { Button } from '@/components/ui/button';
import React, { useState } from 'react'
import { FillAndPreviewPDF } from './ViewPDF';

const ReleasePage = () => {
    const [showPdfViewer, setShowPdfViewer] = useState(false);
    const [view, setView] = useState(true)
    const qaData = {
        "_id": "670cb671eb96e6534cd889da",
        "stage": "ATLAS",
        "status": "Ready for Sign",
        "cequenceId_Hashed": "2119fdc4d0c0444318bfc5a7e8e2ac7c:351e4e7960f9a3a377a0e28e9ae103e5a4aa47f73cb2f7c1ff4f790409444edd9a7fab1b23a029faf789b11f115fa3ea",
        "patientName": "Sanjay",
        "patientDOB": "2024-10-14T06:13:05.105Z",
        "patientWeight": "78",
        "cquenceDIN": "DIN00777",
        "cquenceOrderId": "ORD15",
        "country": "IN",
        "coicBagId": "COIC883",
        "batchNumber": "B114AF",
        "createdAt": "2024-10-14T06:13:05.105Z",
        "updatedAt": "2024-10-14T06:14:55.512Z",
        "__v": 0,
        "pccNumber": "PCC8787",
        "totalVolume": "100",
        "productDose": "10mg",
        "marketAuthorizationNumber": "123564",
        "nameAndAddress": "David Laid Seattle WA 90878",
        "productNDC": "123",
        "expirationDate": "2024-10-14T06:13:05.105Z"
    }



    const handleViewPDF = () => {
        console.log("in handle view");
        setView(false)
        setShowPdfViewer(true);
    };
    return (
        <div>
            <div style={{ display: 'flex', flexDirection: 'column', gap: '20px', justifyContent: 'space-between' }}>
                {view &&
                    <Button onClick={handleViewPDF} style={{ width: '200px', display: 'flex', alignSelf: 'center', marginTop: '10px', justifyContent: 'center' }}>
                        View PDF
                    </Button>
                }
                {
                    showPdfViewer && (
                        <div style={{ display: 'flex', alignSelf: 'center', marginTop: '20px', justifyContent: 'center' }}>
                            <FillAndPreviewPDF

                                data={qaData}

                            />
                        </div>
                    )
                }

            </div >
        </div>
    )
}

export default ReleasePage
