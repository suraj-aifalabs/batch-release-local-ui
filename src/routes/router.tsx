import { createBrowserRouter, Navigate } from 'react-router-dom'
import Layout from '../pages/Layout'
import LandingPage from '../pages/LandingPage'
import ReleasePage from '../pages/release_page/ReleasePage'
import QCManagement from '../pages/config/QCManagement'

export const router = createBrowserRouter([
    {
        path: "/",
        element: <Navigate to="/setting" replace />
    },
    {
        path: "/setting",
        element: <Layout />,
        children: [
            { index: true, element: <LandingPage /> },
            { path: "release", element: <ReleasePage /> },
            { path: "config", element: <QCManagement /> },
        ]
    }
])