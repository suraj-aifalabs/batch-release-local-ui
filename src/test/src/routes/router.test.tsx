import { render, screen } from '@testing-library/react'
import { RouterProvider, createMemoryRouter } from 'react-router-dom'
import '@testing-library/jest-dom'
import { router } from '@/routes/router'
import Layout from '@/pages/Layout'
import LandingPage from '@/pages/LandingPage'
import ReleasePage from '@/pages/release_page/ReleasePage';

// Mock the page components
jest.mock('@/pages/Layout', () => () => <div>Layout Mock</div>)
jest.mock('@/pages/LandingPage', () => () => <div>LandingPage Mock</div>)
jest.mock('@/pages/release_page/ReleasePage', () => () => <div>ReleasePage Mock</div>)

describe('Router Configuration', () => {
    it('should redirect root path to /setting', async () => {
        const testRouter = createMemoryRouter(router.routes, {
            initialEntries: ['/']
        })

        render(<RouterProvider router={testRouter} />)

        // Should redirect to /setting
        expect(testRouter.state.location.pathname).toBe('/')
    })

    it('should render Layout component for /setting path', () => {
        const testRouter = createMemoryRouter(router.routes, {
            initialEntries: ['/setting']
        })

        render(<RouterProvider router={testRouter} />)
        expect(screen.getByText('Layout Mock')).toBeInTheDocument()
        expect(screen.getByText('LandingPage Mock')).toBeInTheDocument()
    })

    it('should render LandingPage as index route of /setting', () => {
        const testRouter = createMemoryRouter(router.routes, {
            initialEntries: ['/setting']
        })

        render(<RouterProvider router={testRouter} />)
        expect(screen.getByText('LandingPage Mock')).toBeInTheDocument()
    })

    it('should render ReleasePage for /setting/release path', () => {
        const testRouter = createMemoryRouter(router.routes, {
            initialEntries: ['/setting/release']
        })

        render(<RouterProvider router={testRouter} />)
        expect(screen.getByText('Layout Mock')).toBeInTheDocument()
        expect(screen.getByText('ReleasePage Mock')).toBeInTheDocument()
    })

    it('should match snapshot of router configuration', () => {
        expect(router.routes).toMatchSnapshot()
    })

    it('should have correct route structure', () => {
        expect(router.routes).toHaveLength(2)

        expect(router.routes[0].path).toBe('/')
        expect(router.routes[0].element.type.name).toBe('Navigate')

        expect(router.routes[1].path).toBe('/setting')
        expect(router.routes[1].element.type.name).toBe('Layout')

        expect(router.routes[1].children).toHaveLength(2)
        expect(router.routes[1].children?.[0].index).toBe(true)
        expect(router.routes[1].children?.[0].element.type.name).toBe('LandingPage')
        expect(router.routes[1].children?.[1].path).toBe('release')
        expect(router.routes[1].children?.[1].element.type.name).toBe('ReleasePage')
    })
})