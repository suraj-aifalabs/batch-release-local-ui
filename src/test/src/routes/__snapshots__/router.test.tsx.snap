// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router Configuration should match snapshot of router configuration 1`] = `
[
  {
    "children": undefined,
    "element": {
      "$$typeof": Symbol(react.transitional.element),
      "_owner": null,
      "_store": {},
      "key": null,
      "props": {
        "replace": true,
        "to": "/setting",
      },
      "type": [Function],
    },
    "hasErrorBoundary": false,
    "id": "0",
    "path": "/",
  },
  {
    "children": [
      {
        "element": {
          "$$typeof": Symbol(react.transitional.element),
          "_owner": null,
          "_store": {},
          "key": null,
          "props": {},
          "type": [Function],
        },
        "hasErrorBoundary": false,
        "id": "1-0",
        "index": true,
      },
      {
        "children": undefined,
        "element": {
          "$$typeof": Symbol(react.transitional.element),
          "_owner": null,
          "_store": {},
          "key": null,
          "props": {},
          "type": [Function],
        },
        "hasErrorBoundary": false,
        "id": "1-1",
        "path": "release",
      },
    ],
    "element": {
      "$$typeof": Symbol(react.transitional.element),
      "_owner": null,
      "_store": {},
      "key": null,
      "props": {},
      "type": [Function],
    },
    "hasErrorBoundary": false,
    "id": "1",
    "path": "/setting",
  },
]
`;
